lib/addons/secrets-store/csi-driver-provider-aws-secrets.ts

```diff
@@ -0,0 +1,195 @@
+      const data: KubernetesSecretObjectData[] = [];
+      let kubernetesSecret: KubernetesSecret;
+      let secretName: string;
+      const secret: ISecret | IStringParameter = teamSecret.secretProvider.provide(clusterInfo); 
+
+      if (Object.hasOwnProperty.call(secret, 'secretArn')) {
+        const secretManagerSecret = secret as ISecret;
+        secretName = secretManagerSecret.secretName;
+        this.parameterObjects.push({
+          objectName: secretManagerSecret.secretName,
+          objectType: AwsSecretType.SECRETSMANAGER
+        });
+        secretManagerSecret.grantRead(serviceAccount);
+      }
+      else {
+        const ssmSecret = secret as IStringParameter;
+        secretName = ssmSecret.parameterName;
+        this.parameterObjects.push({
+          objectName: ssmSecret.parameterName,
+          objectType: AwsSecretType.SSMPARAMETER
+        });
+        ssmSecret.grantRead(serviceAccount);
+      }
+
+      if (teamSecret.kubernetesSecret) {
+        if (teamSecret.kubernetesSecret.data) {
+          teamSecret.kubernetesSecret.data.forEach ( (item) => {
+            const dataObject: KubernetesSecretObjectData = {
+              objectName: item.objectName ?? secretName,
+              key: item.key ?? secretName
+            }
+            data.push(dataObject);
+          });
+        }
+        else {
+          const dataObject: KubernetesSecretObjectData = {
+            objectName: secretName,
+            key: secretName
+          }
+          data.push(dataObject);
+        }
+        kubernetesSecret = {
+          secretName: teamSecret.kubernetesSecret.secretName,
+          type: teamSecret.kubernetesSecret.type ?? KubernetesSecretType.OPAQUE,
+          labels: teamSecret.kubernetesSecret.labels ?? undefined,
+          data,
+        }
+        this.kubernetesSecrets.push(kubernetesSecret);
+      }
+    });
+  }
+
+  /**
+   * Create and apply the SecretProviderClass manifest
+   * @param clusterInfo
+   * @param team
+   * @param csiDriver
+   */
+  private createSecretProviderClass(clusterInfo: ClusterInfo, team: ApplicationTeam, csiDriver: Construct) {
+    const cluster = clusterInfo.cluster;
+    const secretProviderClass = team.teamProps.name + '-aws-secrets';
+    const secretProviderClassManifest = cluster.addManifest(secretProviderClass, {
+      apiVersion: 'secrets-store.csi.x-k8s.io/v1alpha1',
+      kind: 'SecretProviderClass',
+      metadata: {
+        name: secretProviderClass,
+        namespace: team.teamProps.namespace
+      },
+      spec: {
+        provider: 'aws',
+        parameters: {
+          objects: JSON.stringify(this.parameterObjects),
+        },
+        secretObjects: this.kubernetesSecrets
+      }
+    });
+
+    secretProviderClassManifest.node.addDependency(
+      team.serviceAccount,
+      csiDriver
+    );
+
+    new CfnOutput(clusterInfo.cluster.stack, `team-${team.teamProps.name}-secret-provider-class `, {
```

Comment:
This is great, can you add a small section to the doc with the expected output so that admins will know what to hand over to the dev team?

Reply:
added.